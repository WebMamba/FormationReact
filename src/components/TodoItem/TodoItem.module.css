.container {
  padding: 12px;
  margin: 16px;
  border-radius: 14px;
  box-shadow: 0px 4px 5px -2px rgba(0, 0, 0, 0.2),
    0px 7px 10px 1px rgba(0, 0, 0, 0.14), 0px 2px 16px 1px rgba(0, 0, 0, 0.12);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.title {
  font-family: "Montserrat", sans-serif;
  font-weight: 700;
  font-size: 1.2em;
  margin: 0;
  color: #568083;
}

.caption {
  font-family: "Montserrat", sans-serif;
  font-weight: 300;
  font-size: 0.8em;
  font-style: italic;
  margin: 0;
  color: #95c5c9;
}

.checked {
  border-bottom: #c94288 4px solid;
  box-shadow: 0px 4px 5px -2px rgba(219, 22, 193, 0.2),
    0px 7px 10px 1px rgba(219, 22, 193, 0.2),
    0px 2px 16px 1px rgba(219, 22, 193, 0.2);
}
