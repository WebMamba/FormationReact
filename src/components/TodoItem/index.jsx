import styles from "./TodoItem.module.css";
import CheckBox from "../CheckBox";
import { useState } from "react";

export default function TodoItem(props) {
  const { title, datetime } = props;

  const [checked, setChecked] = useState(false);

  const handleCheck = () => {
    setChecked(!checked);
  };

  return (
    <div
      data-testid="todo-item"
      className={`${styles.container} ${checked ? styles.checked : ""}`}
    >
      <div>
        <h5 className={styles.title}>{title}</h5>
        <p className={styles.caption}>
          Ajouté le {datetime.getDay()}/{datetime.getMonth()} à{" "}
          {datetime.getHours()}h
        </p>
      </div>
      <CheckBox checked={checked} onCheck={handleCheck} />
    </div>
  );
}
