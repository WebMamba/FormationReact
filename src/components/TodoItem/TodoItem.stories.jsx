import TodoItem from ".";

export default {
  title: "Components/TodoItem",
  component: TodoItem,
};

export const NotValided = () => (
  <TodoItem title="Faire une react App" datetime={new Date()} />
);
