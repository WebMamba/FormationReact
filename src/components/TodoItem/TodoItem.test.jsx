import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TodoItem from ".";

test("render TodoItem without crashing", () => {
  const component = render(
    <TodoItem title="Test" datetime={new Date("2/1/22")} />
  );

  expect(component).toMatchSnapshot();
});

test("click on checkbox change css class", () => {
  render(<TodoItem title="Test" datetime={new Date("2/1/22")} />);

  const checkBox = screen.getByRole("button");
  const container = screen.getByTestId("todo-item");

  userEvent.click(checkBox);

  expect(checkBox).toHaveClass("check");
  expect(container).toHaveClass("checked");
});
