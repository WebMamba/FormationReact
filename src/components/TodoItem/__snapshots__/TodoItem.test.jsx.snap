// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TodoItem without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container "
        data-testid="todo-item"
      >
        <div>
          <h5
            class="title"
          >
            Test
          </h5>
          <p
            class="caption"
          >
            Ajouté le 
            2
            /
            1
             à
             
            0
            h
          </p>
        </div>
        <button
          class="checkbox not_check"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container "
      data-testid="todo-item"
    >
      <div>
        <h5
          class="title"
        >
          Test
        </h5>
        <p
          class="caption"
        >
          Ajouté le 
          2
          /
          1
           à
           
          0
          h
        </p>
      </div>
      <button
        class="checkbox not_check"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
